compiler/packages/babel-plugin-react-compiler/src/HIR/PrintHIR.ts

```diff
@@ -892,7 +892,8 @@ export function printType(type: Type): string {
   if (type.kind === 'Object' && type.shapeId != null) {
     return `:T${type.kind}<${type.shapeId}>`;
   } else if (type.kind === 'Function' && type.shapeId != null) {
-    return `:T${type.kind}<${type.shapeId}>`;
+    const returnType = printType(type.return);
+    return `:T${type.kind}<${type.shapeId}>()${returnType !== '' ? returnType : ''}`;
```

Comment:
What about something like this to make it even clearer its a return type?

```suggestion
    return `:T${type.kind}<${type.shapeId}>() => ${returnType !== '' ? returnType : ''}`;
```

Reply:
landed something similar, good idea!